### DISCUSSION TITLE
Bulk & Batch Support

### DISCUSSION THREAD

**Discussion by mumby0168** [IEvangelist/azure-cosmos-dotnet-repository#116] [Ideas]
I have just been watching this video 

https://channel9.msdn.com/Shows/Azure-Friday/How-to-do-Bulk-and-Transactional-Batch-operations-with-the-Azure-Cosmos-DB-NET-SDK

Really got me thinking, we already seem as tho we support bulk, I think if we turn the feature on in the`ComosClient` and then group up all our tasks then this will work, would need to verify this.

Batch however is really cool and would be interesting to look at, what do people think?

**Comment by mumby0168** [comment]
Some more information on this here https://devblogs.microsoft.com/cosmosdb/introducing-bulk-support-in-the-net-sdk/

**Comment by IEvangelist** [comment]
Hi @mumby0168 - yes, there has been an option for this since this thing was created:

https://github.com/IEvangelist/azure-cosmos-dotnet-repository/blob/64271f6c44af6be6170073b663c09e209b38db2d/Microsoft.Azure.CosmosRepository/src/Options/RepositoryOptions.cs#L66

And, it is assigned here:

https://github.com/IEvangelist/azure-cosmos-dotnet-repository/blob/64271f6c44af6be6170073b663c09e209b38db2d/Microsoft.Azure.CosmosRepository/src/Providers/DefaultCosmosClientOptionsProvider.cs#L52

**Reply by mumby0168** [reply]
Ahh, I see didn't realize! Does the SDK still provide the same support with it been a `ValueTask<T>` return type I had seen you have to use `Task.WhenAll(valueTasks.Select(vt => vt.AsTask())` to tell the cosmos SDK to execute this as a bulk operation. Just wondered if that had any impact on the way they have implemented it? Can't imagine it does.

**Reply by IEvangelist** [reply]
It shouldn't matter, with my testing of it earlier on - it appeared to work correctly. They internally track invocations, in fact their code is all open source - you can look at how they're doing it.
